charts/linkerd2/templates/web.yaml

```diff
@@ -67,7 +67,8 @@ spec:
         - -log-level={{.Values.controllerLogLevel}}
         {{- $hostFull := replace "." "\\." (printf "linkerd-web.%s.svc.%s" .Values.global.namespace .Values.global.clusterDomain) }}
         {{- $hostAbbrev := replace "." "\\." (printf "linkerd-web.%s.svc" .Values.global.namespace) }}
-        - -enforced-host=^(localhost|127\.0\.0\.1|{{ $hostFull }}|{{ $hostAbbrev }}|\[::1\])(:\d+)?$
+        {{- $default_host :=^(localhost|127.0.0.1| $hostFull | $hostAbbrev |[::1])(:\d+)?$ }}
+        - -enforced-host={{ .Values.global.enforcedHostRegexp | default $default_host }}
```

Comment:
I think the code here can be written as
```
        {{- if .Values.enforcedHostRegexp }}
        - -enforced-host= {{.Values.enforcedHostRegexp}}
        {{- else -}}
        {{- $hostFull := replace "." "\\." (printf "linkerd-web.%s.svc.%s" .Values.global.namespace .Values.global.clusterDomain) }}
        {{- $hostAbbrev := replace "." "\\." (printf "linkerd-web.%s.svc" .Values.global.namespace) }}
        - -enforced-host=^(localhost|127\.0\.0\.1|{{ $hostFull }}|{{ $hostAbbrev }}|\[::1\])(:\d+)?$
        {{- end}}
```
There should be a better way to do this using `default` but I kept getting syntax errors. because of `^`, etc symbols, enclosing it into a string using `printf` did not help :\ @alpeb

Reply:
Thanks @Pothulapati , will update this and push again